samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ConfirmPublishAssessmentListener.java

```diff
@@ -396,9 +396,16 @@ else if (time instanceof java.lang.Boolean) {
 				String date_err=ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.AssessmentSettingsMessages","date_error");
 				context.addMessage(null,new FacesMessage(date_err));
 			}
+			else {
+				if(StringUtils.isNotBlank(assessmentSettings.getFeedbackEndDateString()) && assessmentSettings.getFeedbackDate().after(assessmentSettings.getFeedbackEndDate())){
```

Comment:
can getFeedbackDate return null here? In the last PR it looked like that was the concern, even though the logic was the wrong way round.This whole approach seems odd, tbh. Why are we checking two different things? It would be a lot better to just have one definitive thing that shows that there has been feedback, then test that. I'm not saying you've made it strange - it was probably strange to begin with. Could you standardise on one test, either getFeedbackDate, or getFeedbackDateString? Or even add an hasFeedback flag to the assessment settings object.

How about using getIsValidFeedbackDate as your universal test?

Reply:
getFeedbackDate can't be null here. 
If getFeedbackDate was null then the code enters on the if Line 394 and skip the else.

getFeedbackDate and getFeedbackDateString are the same. 
I don't know why sometimes was programming with getFeedbackDate and other with getFeedbackDateString.
(I haven't changed that), I have only ordered that lines (if and else) because and NPE appears.